crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -434,10 +435,25 @@ impl Inner {
         let rows = st_sequences.scan_rows().cloned().collect::<Vec<_>>();
         for row in rows {
             let sequence = StSequenceRow::try_from(&row)?;
+            // TODO: The system tables have initialized their value already, but this is wrong:
+            // If we exceed  `SEQUENCE_PREALLOCATION_AMOUNT` we will get a unique violation
+            let is_system_table = self
+                .committed_state
+                .tables
+                .get(&TableId(sequence.table_id))
+                .map_or(false, |x| x.schema.table_type == StTableType::System);
+
             let schema = (&sequence).into();
+
+            let mut seq = Sequence::new(schema);
+            // Now we need to recover the last allocation value.
+            if !is_system_table && seq.value < sequence.allocated + 1 {
```

Comment:
I am not sure this is what we want: when restarting, the next sequence value will effectively be > SEQUENCE_PREALLOCATION_AMOUNT (ie. 4096), instead of value + 1.

Did you mean to check if the value is _greater_ then allocated?

Reply:
Yes, but this is how it works in PG (with a lesser `SEQUENCE_PREALLOCATION_AMOUNT`). Do otherwise demand scan all tables and eventually, this could be expensive (will not if exists a way to scan in reverse).

Reply:
Ok makes sense now